packages/client/hmi-client/src/components/data-explorer/search-item.vue

```diff
@@ -17,18 +17,24 @@ const props = defineProps<{
 const emit = defineEmits(['toggle-selected-asset', 'toggle-asset-preview']);
 
 const relatedAssetPage = ref<number>(0);
+const chosenExtractionFilter = ref<XDDExtractionType | null>(null);
 
 // These asset types don't appear at the moment
 const extractionsWithImages = computed(() =>
 	props.asset.relatedExtractions
-		? props.asset.relatedExtractions?.filter(
-				(ex) =>
-					ex.askemClass === XDDExtractionType.Figure ||
-					ex.askemClass === XDDExtractionType.Table ||
-					ex.askemClass === XDDExtractionType.Equation ||
-					ex.askemClass === XDDExtractionType.Section || // remove this later just for testing pagination
-					ex.askemClass === XDDExtractionType.Document // remove this later just for testing pagination
-		  )
+		? props.asset.relatedExtractions?.filter((ex) => {
+				if (chosenExtractionFilter.value === null) {
+					return (
+						ex.askemClass === XDDExtractionType.Figure ||
+						ex.askemClass === XDDExtractionType.Table ||
+						ex.askemClass === XDDExtractionType.Equation ||
+						ex.askemClass === XDDExtractionType.Section || // remove this later just for testing pagination
+						ex.askemClass === XDDExtractionType.Document || // remove this later just for testing pagination
+						ex.askemClass === undefined // remove this later just for testing pagination
+					);
```

Comment:
Why is there an extra `undefined` clause now? and the associated comments don't seem to make any sense??

Anyway, if we can exclude needing `undefined` you can probably do something like this

```
return ex.askemClass in XDDExtractionType
```

Reply:
xdd has been returning undefined classes as of late. It's just there so I can test the pagination. Once Derek fixes the backend stuff I'll only be filtering through table, figure, url, equation